src/runtime/divert.rs

```diff
@@ -97,36 +106,24 @@ enum DivertType {
 
 impl From<DivertData> for Divert {
     fn from(divert: DivertData) -> Self {
-        match divert.divert_type {
-            DivertType::Standard { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new()
-            },
-            DivertType::Variable { target, var: () } => Self {
-                target: Some(TargetType::VarName(target)),
-                is_conditional: divert.conditional,
-                ..Self::new()
-            },
-            DivertType::Function { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new_function()
-            },
-            DivertType::Tunnel { path } => Self {
-                target: Some(TargetType::Path(path)),
-                is_conditional: divert.conditional,
-                ..Self::new_tunnel()
-            },
+        let new_divert = match divert.divert_type {
+            DivertType::Standard { path } => Self::new_divert(TargetType::Path(path)),
+            DivertType::Variable { target, var: () } => {
+                Self::new_divert(TargetType::VarName(target))
+            }
+            DivertType::Function { path } => Self::new_function(TargetType::Path(path)),
+            DivertType::Tunnel { path } => Self::new_tunnel(TargetType::Path(path)),
             DivertType::ExternalFunction {
                 external_func_name,
                 external_arguments,
-            } => Self {
-                target: Some(TargetType::ExternalName(external_func_name)),
-                external_args: Some(external_arguments),
-                is_conditional: divert.conditional,
-                ..Self::new_external_function()
-            },
+            } => Self::new_external_function(
+                TargetType::ExternalName(external_func_name),
+                external_arguments,
+            ),
+        };
+        Self {
+            is_conditional: divert.conditional,
+            ..new_divert
         }
```

Comment:
I have to look into this syntax a little bit but I'm somewhat worried about it using extra memory. Do you know the name of it so I can look it up?

Reply:
[Struct Update Syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) is the name. [StackOverflow](https://stackoverflow.com/q/48212841) is inconclusive about whether it's optimised in that way. Since the constructor is in the same crate I'd _expect_ it to get inlined and optimised away at compile time. And there's no way it would allocate memory even optimisation didn't happen, it would just use the stack.
But it's moot if removing the constructors.